ow_lander/scripts/all_action_trajectories.py

```diff
@@ -856,8 +855,45 @@ def deliver_sample(move_arm, robot, moveit_fk, args):
     if len(plan_b.joint_trajectory.points) == 0:  # If no plan found, send the previous plan only
         return plan_a
 
+    discard_sample_traj = cascade_plans(plan_a, plan_b)
+
+    return discard_sample_traj
+
+def deliver_sample(move_arm, robot, moveit_fk):
+    """
+    :type move_arm: class 'moveit_commander.move_group.MoveGroupCommander'
+    """
+    move_arm.set_planner_id("RRTstar")
+    robot_state = robot.get_current_state()
+    move_arm.set_start_state(robot_state)
+    goal = move_arm.get_current_pose().pose
+    goal = move_arm.get_named_target_values("arm_deliver_staging_1")
+    move_arm.set_joint_value_target(goal)
+    _, plan_a, _, _ = move_arm.plan()
+    if len(plan_a.joint_trajectory.points) == 0:  # If no plan found, abort
+        return False
+
+    cs, start_state, goal_pose = calculate_joint_state_end_pose_from_plan_arm(
+        robot, plan_a, move_arm, moveit_fk)
+    move_arm.set_start_state(cs)
+
+    goal = move_arm.get_named_target_values("arm_deliver_staging_2")
+    # move_arm.set_pose_target(goal_pose)
+    move_arm.set_joint_value_target(goal)
+    _, plan_b, _, _ = move_arm.plan()
+    if len(plan_b.joint_trajectory.points) == 0:  # If no plan found, send the previous plan only
+        return plan_a
     deliver_sample_traj = cascade_plans(plan_a, plan_b)
 
-    # move_arm.set_planner_id("RRTconnect")
+    cs, start_state, goal_pose = calculate_joint_state_end_pose_from_plan_arm(
+        robot, deliver_sample_traj, move_arm, moveit_fk)
+    move_arm.set_start_state(cs)
+    goal = move_arm.get_named_target_values("arm_deliver_final")
+    move_arm.set_joint_value_target(goal)
+    _, plan_c, _, _ = move_arm.plan()
```

Comment:
It should be possible to re-factor this code and iterate over the predefined pose using a simple for loop in a similar manner to the code of the [equivalent ros service](https://github.com/nasa/ow_simulator/blob/0ad1d797cb2e02a96da738f57c9578d965b5215b/ow_lander/scripts/path_planning_commander.py#L84) to avoid code blocks duplications.

Reply:
> It should be possible to re-factor this code and iterate over the predefined poses using a simple for loop in a similar manner to the code of the [equivalent ros service](https://github.com/nasa/ow_simulator/blob/0ad1d797cb2e02a96da738f57c9578d965b5215b/ow_lander/scripts/path_planning_commander.py#L84) to avoid repeating the same code block.

Done!! But the for loop only has the last 2 points and not all the three. The starting position of the first part of the plan comes from the lander but the starting position of the last two is derived from the plan.

Reply:
Please review #215

Reply:
Done.. merged #215